### PR TITLE
Assignment 7 Submission

### PR DESCRIPTION
**Pull request by raymondshiner**
Ready for reviews when people want to. But I'm still working on it.

### CODE REVIEW

File: SecretSanta/src/SecretSanta.Web/Views/Users/Add.cshtml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿@model UserInputViewModel
+@if (!string.IsNullOrWhiteSpace(ViewBag.ErrorMessage))
+{
+    <div>@ViewBag.ErrorMessage</div>
+}
+<form asp-route="Add">
+    <div class="field">
+        <label class="label" asp-for="FirstName">First Name</label>
+        <div class="control">
+            <input class="input" type="text" asp-for="FirstName" />
```
Comment (RichardTeller) [inline SecretSanta/src/SecretSanta.Web/Views/Users/Add.cshtml]: Remeber in final submission to check for valid field

File: SecretSanta/src/SecretSanta.Web/Views/Groups/Add.cshtml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿@model GroupInputViewModel
+
+<div asp-validation-summary="All"></div>
```
Comment (RichardTeller) [inline SecretSanta/src/SecretSanta.Web/Views/Groups/Add.cshtml]: I cant remember correctly if just this tag checks for validity.  I think you may still need the asp-validation-for tag on the Name input.  Something to note

File: SecretSanta/src/SecretSanta.Web/Views/Groups/Edit.cshtml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿@model GroupViewModel
+
+<div asp-validation-summary="All"></div>
```
Comment (RichardTeller) [inline SecretSanta/src/SecretSanta.Web/Views/Groups/Edit.cshtml]: Same as previous comment

File: SecretSanta/src/SecretSanta.Web/Controllers/GroupsController.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+        }
+
+        public async Task<IActionResult> Delete(int id)
+        {
+            IActionResult result = null;
+            using (var httpClient = ClientFactory.CreateClient("SecretSantaApi"))
+            {
+                try
+                {
+                    var blogEngineClient = new SecretSantaClient(httpClient.BaseAddress.ToString(), httpClient);
```
Comment (RichardTeller) [inline SecretSanta/src/SecretSanta.Web/Controllers/GroupsController.cs]: blogEngineClient??? COPY PASTA!?!?!

Comment (raymondshiner) [reply SecretSanta/src/SecretSanta.Web/Controllers/GroupsController.cs]: Well Drat you found me out. lol

**Comment by breaman** [comment]
Nice work on this assignment, you added some polish that others missed and it made the experience just a little bit nicer. One thing that was missing, though, was when trying to add a user without a first name, it failed, but the application didn't tell the person why it failed.

**Pull request closed without merging**